src/parser/parser.ts

```diff
@@ -0,0 +1,237 @@
+import {
+  CasperServiceByJsonRPC,
+  decodeBase16,
+  encodeBase16,
+} from 'casper-js-sdk';
+import {
+  parseBytesWithRemainder,
+  parseCLValueFromBytesWithRemainder,
+} from './casper/utils';
+import {
+  Event,
+  parseEventDataFromBytes,
+  parseEventNameWithRemainder,
+} from './event';
+import { ExecutionResult, RawCLValue, WithRemainder } from './casper/types';
+import { parseSchemasFromBytes, Schemas } from './schema';
+
+export interface ContractMetadata {
+  schemas: Schemas;
+  contractHash: Uint8Array;
+  contractPackageHash: Uint8Array;
+  eventsSchemaUref: string;
+  eventsUref: string;
+}
+
+export const EVENTS_SCHEMA_NAMED_KEY = '__events_schema';
+
+export const EVENTS_NAMED_KEY = '__events';
+
+const DICTIONARY_PREFIX = 'dictionary-';
+
+export interface ParseResult {
+  error: string | null;
+  event: Event;
+}
+
+export class Parser {
+  constructor(
+    private readonly contractsMetadata: Record<string, ContractMetadata>,
+  ) {}
+
+  static async initialize(
```

Comment:
Since it's factory method then may let's give it a corresponding name, like `newParser`, `makeParser`,  `createParser`?